### ISSUE TITLE
Unique index for booking option identifier

### ISSUE THREAD

**Issue by bernhard-wunderbyte** [Wunderbyte-GmbH/moodle-mod_booking#300]
We need to make sure that booking option identifiers are truly unique.

1. Create a **migration script** to change duplicate identifiers to unique identifiers - use the function booking_option::create_truly_unique_option_identifier() for identifier creation!

2. Create a **UNIQUE index** for column "identifier" of table "booking_options".

**Comment by georgmaisser** [comment]
I guess this was done a while ago.

**Comment by bernhard-wunderbyte** [comment]
@georgmaisser This was not yet done as there would be some implications to it which still need discussion. I remember a conversation with @dasistwas where we were discussing if identifiers should really be "truly unique" or only unique per instance. There would be both advantages (easy migration of options to another booking instance) and disadvantages (serious danger of confusing booking options with each other and of overwriting). As we did not decide the final design yet, we also decided to not yet implement a migration script and a unique key. However the function booking_option::create_truly_unique_option_identifier()  currently makes sure that new identifiers are truly unqiue on the whole Moodle site. I'll re-open this until we have decided on the final design.
